.github/workflows/reusable-macos.yml

```diff
@@ -47,11 +47,23 @@ jobs:
           ${{ inputs.free-threading && '--disable-gil' || '' }} \
           --prefix=/opt/python-dev \
           --with-openssl="$(brew --prefix openssl@3.0)"
+    - name: Build CPython for Compiler Warning Check
+      if : ${{ !inputs.free-threading && inputs.os == 'macos-14' }}
+      run: set -o pipefail; make -j1 2>&1 | tee compiler_output_macos.txt
```

Comment:
Why `-j1` here? How long does it take compared with `-j4` or `-j8`?

Reply:
I found that with the macOS job there is a race condition between the threads to write out compiler output which would lead to malformed warnings. By using `-j1` this eliminates that issue

Reply:
@hugovk I pushed a commit that builds with `-j8` for the warning check build. You can see here that the output is malformed: https://github.com/python/cpython/actions/runs/10406271367/job/28818954178?pr=123020. I am looking into seeing if clang has an option to sync output

Reply:
Turns out I can use home-brew in the macOS job to install make, then use gmake instead of the system BSD make to build. Then the option `--output-sync` can be enabled so that writing to stdout is thread safe. I put this in place and reverted to `-j8` and it looks like there are no issues in compiler output.

Reply:
Ah, right.

@erlend-aasland Do you think it's okay to use Homebrew's make here?

Reply:
The race condition is within make itself. When you build using multiple threads (`-j8`) each of those threads races to write the diagnostics of the build to stdout and  the make output itself is jumbled. GNU make has an option to sync the output from the threads so that it is legible `--output-sync`.